src/gluonts/zebras/_schema.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+from dataclasses import dataclass
+from typing import Any, Callable, Optional, Union, Type, Dict
+
+import numpy as np
+from toolz import valfilter
+
+from gluonts.itertools import partition
+
+from ._freq import Freq
+from ._period import Period
+from ._time_frame import time_frame, TimeFrame
+from ._split_frame import split_frame, SplitFrame
+
+"""
+This module provides tooling to extract ``zebras.TimeFrame`` and
+``zebras.SplitFrame`` instances from Python dictionaries::
+
+    schema = zebras.Schema(...)
+    tf = schema.load_timeframe(raw_dict)
+
+    sf = schema.load_splitframe(raw_dict)
+
+The idea is to define expected types and shapes for each field (column) using
+``zebras.Field`` and then wrap them in a ``zebras.Schema``:
+
+    schema = zebras.Schema({
+        "target": Field(ndim=1, tdim=0, past_only=True),
+        "time_feat": zebras.Field(ndim=2, tdim=-1),
+        "static_feat": zebras.Field(ndim=1),
+
+    })
+
+See the documentation of ``Field`` on how to configure the columns of a schema.
+"""
+
+
+@dataclass
+class Field:
+    """
+    Specification for user provided input data.
+
+    Parameters
+    ----------
+    ndim, optional
+        The expected number of dimensions of the input data. If provided, it is
+        ensured that the input array has the expected number of dimensions.
```

Comment:
What happens if it's not provided? Is this validation just skipped, or is the argument instead assumed to be a scalar (non-array)?

Reply:
Just skipped.